clowder-framework/event-sink-consumers
# Add InfluxDB event consumer

NOTE: This will work with `InfluxDB 1.8.3`, but will need modifications to work with their new 2.0 release

## How to Test
Prerequisite: RabbitMQ running locally

1. Start up an InfluxDB instance:
```bash
docker run -itd --name=influxdb -p 8086:8086 -v $PWD/influxdb:/var/lib/influxdb influxdb:1.8.3
```
2. Start up one or more InfluxDB consumer and send a test message:
    * Terminal A: `./recv.sh`
    * Terminal B: `./send.sh`

3. You should then be able to query InfluxDB to see the messages:
```bash
$ docker exec -it influxdb bash
root@0da139d423aa:/# influx
Connected to http://localhost:8086 version 1.8.3
InfluxDB shell version: 1.8.3
> show databases
name: databases
name
----
eventsink
_internal
> use eventsink
Using database eventsink
> SELECT * from events;
name: events
time          message      resourceId                           user
----          -------      ----------                           ----
1610742782580 Hello World! 6c89f539-71c6-490d-a28d-6c5d84c0ee2f User
1610742808889 Hello World! 6c89f539-71c6-490d-a28d-6c5d84c0ee2f User
1610742812964 Hello World! 6c89f539-71c6-490d-a28d-6c5d84c0ee2f User
1610742822369 Hello World! 6c89f539-71c6-490d-a28d-6c5d84c0ee2f User  
```
